src/status_im/utils/mixpanel.cljs

```diff
@@ -1,35 +1,56 @@
 (ns status-im.utils.mixpanel
   (:require-macros [status-im.utils.slurp :as slurp])
-  (:require [cljs.reader :as reader]
+  (:require [cljs.core.async :as async]
+            [cljs.reader :as reader]
             [goog.crypt.base64 :as b64]
+            [taoensso.timbre :as log]
             [status-im.utils.build :as build]
             [status-im.utils.config :as config]
             [status-im.utils.datetime :as datetime]
             [status-im.utils.http :as http]
             [status-im.utils.platform :as platform]
-            [status-im.utils.types :as types]))
+            [status-im.utils.types :as types]
+            [status-im.react-native.js-dependencies :as rn-dependencies]))
 
 (def base-url "http://api.mixpanel.com/")
 (def base-track-url (str base-url "track/"))
 
 (defn encode [m]
   (b64/encodeString (types/clj->json m)))
 
-(defn- build-url [id label props]
-  (str base-track-url
-       "?data="
-       (encode {:event      label
-                :properties (merge
-                              {:token       config/mixpanel-token
-                               :distinct_id id
-                               :os          platform/os
-                               :os-version  platform/version
-                               :app-version build/version
-                               :time        (datetime/timestamp)}
-                              props)})))
+(defn- make-event [id label props]
+  {:event      label
+   :properties (merge
+                {:token       config/mixpanel-token
+                 :distinct_id id
+                 :os          platform/os
+                 :os-version  platform/version
+                 :app-version build/version
+                 :time        (datetime/timestamp)}
+                props)})
+
+(def ^:private pending-events-queue (async/chan 2000))
+
+(async/go-loop [accumulator []]
+  (let [accumulated (conj accumulator (<! pending-events-queue))
+        next-accumulator (async/chan)]
+    (-> (rn-dependencies/fetch
+         base-track-url
+         (clj->js {:method  "POST"
+                   :body    (str "data=" (encode (into [] (take 50 accumulated))))
```

Comment:
i would move `50` to something like `def batch-size` (same with `50` on line 44)

Reply:
Good catch, will do!